### PR TITLE
Add support for mounting custom volumes

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Adds support for mounting custom volumes on the service deployments

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

**Comment by angrybayblade** [comment]
The test failures are addressed here - https://github.com/valory-xyz/open-autonomy/pull/2213

**Review by dvilelaf — APPROVED** [review]

**Review by 0xArdi — APPROVED** [review]

File: autonomy/deploy/generators/kubernetes/base.py [hunk 1]
```diff
@@ -142,12 +161,25 @@ def generate_config_tendermint(self) -> "KubernetesGenerator":
             ]
         )
 
+        pvcs = ""
+        extra_volumes = self.service_builder.service.deployment_config.get(
+            "agent", {}
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply autonomy/deploy/generators/kubernetes/base.py:166]: No, tendermint nodes does not require extra volumes. These volumes are only used by agents

**Review by solarw — APPROVED** [review]
lgtm

File: autonomy/deploy/generators/kubernetes/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (8ball030) [reply autonomy/deploy/generators/kubernetes/base.py:166]: Errrr
One second ser
There is an initial config job when running a 4 node dpeloyment which copies across the tendermint config to the 4 volumes? 

https://github.com/valory-xyz/open-autonomy/blob/6409b517afae8d1b7bb535124fab1d7e7474944c/autonomy/deploy/generators/kubernetes/templates.py#L129

**Comment by 8ball030** [comment]
Think your gonna have a bad time here

File: autonomy/deploy/generators/kubernetes/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (8ball030) [reply autonomy/deploy/generators/kubernetes/base.py:166]: See here where it actioned from the tm container;

https://github.com/valory-xyz/open-autonomy/blob/6409b517afae8d1b7bb535124fab1d7e7474944c/autonomy/deploy/generators/kubernetes/templates.py#L327

[the remark this answers is not in the document]
Comment (8ball030) [reply autonomy/deploy/generators/kubernetes/base.py:166]: however i could have mis understood.

**Pull request merged**